### PR TITLE
Update storage reading and writing

### PR DESCRIPTION
**Pull request by Seonlo99**

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedNextOfKin.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+     */
+    public JsonAdaptedNextOfKin(NextOfKin source) {
+        if (source.isPresent()) {
+            this.nextOfKinName = source.getName().fullName;
+            this.nextOfKinPhone = source.getPhone().value;
+            this.nextOfKinRelationship = source.getRelationship().relationship;
+        } else {
+            this.nextOfKinName = null;
+            this.nextOfKinPhone = null;
+            this.nextOfKinRelationship = null;
```
Comment (EricXiong420) [inline src/main/java/seedu/address/storage/JsonAdaptedNextOfKin.java:47]: Good work! follows the format of other JsonAdaptable objects

File: src/main/java/seedu/address/storage/JsonAdaptedNextOfKin.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                    "Next of Kin Relationship"));
+        } else {
+            if (!Name.isValidName(nextOfKinName)) {
+                throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
+            }
+            if (!Phone.isValidPhone(nextOfKinPhone)) {
+                throw new IllegalValueException(Phone.MESSAGE_CONSTRAINTS);
+            }
+            if (!Relationship.isValidRelationship(nextOfKinRelationship)) {
+                throw new IllegalValueException(Relationship.MESSAGE_CONSTRAINTS);
```
Comment (EricXiong420) [inline src/main/java/seedu/address/storage/JsonAdaptedNextOfKin.java:88]: Good use of defensive coding.

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 3]
```diff
@@ -271,18 +301,28 @@ public Person toModelType() throws IllegalValueException {
...
         final Set<Affiliation> modelAffiliationHistory = new HashSet<>(personAffiliationHistory);
 
         Person generatedPerson = modelRole.generatePerson(modelName,
                 modelPhone, modelEmail, modelAffiliations, modelAffiliationHistory);
+
+        if (generatedPerson instanceof Patient) {
+            final NextOfKin modelNextOfKin = generateNextOfKin(nextOfKin);
+            Patient patient = ((Patient) generatedPerson);
+            patient.setNextOfKin(modelNextOfKin);
+            return patient;
```
Comment (EricXiong420) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:317]: Uses NextOfKin as unique attribute of Patient, good.

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 4]
```diff
@@ -146,7 +150,7 @@ public void toModelType_invalidSpecialisations_throwsIllegalValueException() {
         JsonAdaptedPerson person =
                 new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_ROLE,
                         VALID_AFFILIATIONS, VALID_AFFILIATION_HISTORY,
-                        VALID_SHIFT_DAYS, INVALID_SPECIALISATIONS);
+                        VALID_SHIFT_DAYS, INVALID_SPECIALISATIONS, VALID_NEXT_OF_KIN);
```
Comment (EricXiong420) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:153]: Caught all the lines that needed the adding of VALID_NEXT_OF_KIN

**Review by EricXiong420 — APPROVED** [review]
LGTM, can go ahead and merge.

File: src/main/java/seedu/address/storage/JsonAdaptedNextOfKin.java (hunk 1 above)
Comment (Seonlo99) [reply src/main/java/seedu/address/storage/JsonAdaptedNextOfKin.java:47]: Thanks! :)

File: src/main/java/seedu/address/storage/JsonAdaptedNextOfKin.java (hunk 2 above)
Comment (Seonlo99) [reply src/main/java/seedu/address/storage/JsonAdaptedNextOfKin.java:88]: Thanks! :)

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java (hunk 3 above)
Comment (Seonlo99) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java:317]: Thanks! :)

**Pull request merged**